File: Aigilas/Aigilas/Entities/ComboMarker.cs
﻿using Aigilas.Creatures;
using Aigilas.Skills;
using SPX.Core;
using SPX.Entities;
using Aigilas.Management;
using System.Collections.Generic;
using System;
using SPX.Particles;

namespace Aigilas.Entities
{
    public class ComboMarker:Entity
    {
        private ICreature _parent;
        private int _index;

        public ComboMarker(ICreature source,int elementId,int index)
        {
            Initialize(source.GetLocation(), SpriteType.COMBO_MARKER, Aigilas.EntityType.COMBO_MARKER,ZDepth.ComboMarker);
            _graphic.SetColor(Elements.Colors[elementId]);
            _graphic.SetAlpha(0);
            ParticleEngine.Emit(SPX.Particles.RotateBehavior.GetInstance(), this, _graphic.GetColor());
            _parent = source;
            _index = index;                   
        }

        private static readonly List<Point2> __dMults = new List<Point2>()
        {
            new Point2(-1,0),
            new Point2(0,-1),
            new Point2(1,0)
        };

        public override void Update()
        {
            float dX = (GameManager.SpriteWidth / 16) * __dMults[_index].X;
            float dY = (GameManager.SpriteHeight / 16) * __dMults[_index].Y;
            SetLocation(new Point2(_parent.GetLocation().PosX + dX, _parent.GetLocation().PosY + dY));
        }

        public override void Draw()
        {
            _graphic.Draw();
        }
    }
}

File: aigilas/src/aigilas/entities/ComboMarker.java
package aigilas.entities;

import aigilas.creatures.BaseCreature;
import aigilas.management.SpriteType;
import sps.bridge.DrawDepth;
import sps.bridge.EntityType;
import sps.core.Point2;
import sps.core.Settings;
import sps.entities.Entity;
import sps.particles.ParticleEngine;

import java.util.Arrays;
import java.util.List;

public class ComboMarker extends Entity {
    private final BaseCreature _parent;
    private final int _index;

    public ComboMarker(BaseCreature source, Elements elementId, int index) {
        initialize(source.getLocation(), SpriteType.Combo_Marker, EntityType.Combo_Marker, DrawDepth.ComboMarker);
        _graphic.setColor(elementId.Tint);
        _graphic.setAlpha(0);
        _parent = source;
        ParticleEngine.emit(sps.particles.behaviors.RotateBehavior.getInstance(), _parent, _graphic.getColor());
        _index = index;
    }

    private static final List<Point2> __dMults = Arrays.asList(new Point2(-1, 0), new Point2(0, -1), new Point2(1, 0));

    @Override
    public void update() {
        float dX = (Settings.get().spriteWidth / 16) * __dMults.get(_index).X;
        float dY = (Settings.get().spriteHeight / 16) * __dMults.get(_index).Y;
        setLocation(new Point2(_parent.getLocation().PosX + dX, _parent.getLocation().PosY + dY));
    }

    @Override
    public void draw() {
        _graphic.draw();
    }
}
